Replace `std::optional<ASCIILiteral>` usage with `ASCIILiteral`

k-monsen opened the pull request:
#### 4e2cbcfebd2f422292fd73d912607356a4a6f5d1
<pre>
Replace `std::optional&lt;ASCIILiteral&gt;` usage with `ASCIILiteral`
<a href="https://bugs.webkit.org/show_bug.cgi?id=287270">https://bugs.webkit.org/show_bug.cgi?id=287270</a>
<a href="https://rdar.apple.com/144810137">rdar://144810137</a>

Reviewed by Ryan Reno.

Removed std::optional from uses of ASCIILiteral as it has a
bool() operator and a null state built in. Changed where std:nullopt
were returned to {}, and removed &apos;#include &lt;wtf/StdLibExtras.h&gt;&apos;
where possible.

* Source/WebCore/crypto/gcrypt/GCryptUtilities.cpp:
(WebCore::hashAlgorithmName):
* Source/WebCore/crypto/gcrypt/GCryptUtilities.h:
* Source/WebCore/platform/graphics/FontCache.cpp:
(WebCore::FontCache::alternateFamilyName):
(WebCore::FontCache::cachedFontPlatformData):
* Source/WebCore/platform/graphics/FontCache.h:
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::platformAlternateFamilyName):
* Source/WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:
(WebCore::FontCache::platformAlternateFamilyName):
* Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp:
(WebCore::FontCache::platformAlternateFamilyName):
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
(WebCore::FontCache::platformAlternateFamilyName):

Canonical link: <a href="https://commits.webkit.org/290745@main">https://commits.webkit.org/290745@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/614fcb01670b25768a9ace9b18d06e33c0a2a31b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89994 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/9523 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44894 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94994 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40767 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/92046 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9910 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17878 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69284 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26894 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92995 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/7584 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/81637 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49641 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7311 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/36250 "Found 2 new test failures: fullscreen/full-screen-request-removed.html media/modern-media-controls/invalid-placard/invalid-placard-constrained-metrics.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39901 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77647 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/37060 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96820 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/17182 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/12653 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/78282 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/17438 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/77457 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77482 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19306 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21935 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/20520 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/10376 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/17192 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/22517 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16933 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/20385 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18716 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
```diff
@@ -81,11 +81,11 @@ static std::optional<Vector<uint8_t>> gcryptSign(gcry_sexp_t keySexp, const Vect
     {
         auto shaAlgorithm = hashAlgorithmName(hashAlgorithmIdentifier);
-        if (!shaAlgorithm)
+        if (shaAlgorithm.isNull())
```
You should just be able to use `!shaAlgorithm` for this (and similar usages)

k-monsen replied:
It works, but the compiler gets confused:
In file included from /app/webkit/WebKitBuild/WPE/Release/WebCore/DerivedSources/unified-sources/UnifiedSource-44a97133-4.cpp:7:
/app/webkit/Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp: In function ‘std::optional<WTF::Vector<unsigned char> > WebCore::gcryptSign(gcry_sexp_t, const WTF::Vector<unsigned char>&, CryptoAlgorithmIdentifier, size_t)’:
/app/webkit/Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:84:13: error: ambiguous overload for ‘operator!’ (operand type is ‘WTF::ASCIILiteral’)
   84 |         if (!shaAlgorithm)
      |             ^~~~~~~~~~~~~
/app/webkit/Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:84:13: note: candidate: ‘operator!(bool)’ (built-in)
In file included from /app/webkit/Source/WebCore/crypto/keys/CryptoKeyAlgorithm.h:28,
                 from /app/webkit/Source/WebCore/crypto/keys/CryptoAesKeyAlgorithm.h:28,
                 from /app/webkit/Source/WebCore/crypto/CryptoKey.h:28,
                 from /app/webkit/Source/WebCore/crypto/CryptoAlgorithm.h:29,
                 from /app/webkit/Source/WebCore/crypto/algorithms/CryptoAlgorithmAESCBC.h:28,
                 from /app/webkit/Source/WebCore/crypto/gcrypt/CryptoAlgorithmAESCBCGCrypt.cpp:27,
                 from /app/webkit/WebKitBuild/WPE/Release/WebCore/DerivedSources/unified-sources/UnifiedSource-44a97133-4.cpp:1:
/app/webkit/WebKitBuild/WPE/Release/WTF/Headers/wtf/text/WTFString.h:334:13: note: candidate: ‘bool WTF::operator!(const String&)’

from https://ews-build.webkit.org/#/builders/5/builds/94881/steps/15/logs/errors

k-monsen replied:
I think it is because it can implicit converted to String and there is a 'bool WTF::operator!(const String)'

kmonsen committed:
Replace `std::optional<ASCIILiteral>` usage with `ASCIILiteral`
https://bugs.webkit.org/show_bug.cgi?id=287270
rdar://144810137

Reviewed by Ryan Reno.

Removed std::optional from uses of ASCIILiteral as it has a
bool() operator and a null state built in. Changed where std:nullopt
were returned to {}, and removed '#include <wtf/StdLibExtras.h>'
where possible.

* Source/WebCore/crypto/gcrypt/GCryptUtilities.cpp:
(WebCore::hashAlgorithmName):
* Source/WebCore/crypto/gcrypt/GCryptUtilities.h:
* Source/WebCore/platform/graphics/FontCache.cpp:
(WebCore::FontCache::alternateFamilyName):
(WebCore::FontCache::cachedFontPlatformData):
* Source/WebCore/platform/graphics/FontCache.h:
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::platformAlternateFamilyName):
* Source/WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:
(WebCore::FontCache::platformAlternateFamilyName):
* Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp:
(WebCore::FontCache::platformAlternateFamilyName):
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
(WebCore::FontCache::platformAlternateFamilyName):

Canonical link: https://commits.webkit.org/290745@main
